Adds warning about directories blocked from being shared (#16277)
gravitational/teleport: docs/pages/desktop-access/directory-sharing.mdx

![The "Share Desktop" button](../../img/desktop-access/share.png)

<Notice type="warning">

> For security reasons it's not possible to share certain directories,

Reviewer: ```suggestion
For security reasons the web browser's filesystem access API prohibits you from sharing certain directories,
```

Let's make it clear this is a browser-imposed limitation, not a Teleport one.

> For security reasons, the browser prevents you from sharing certain directories,

Reviewer 2: ```suggestion
For security reasons it's not possible to share certain directories,
```

> including the standard locations for commonly used user directories such as "desktop"
> and "documents", as well as critical directories more typically used by the operating

Reviewer 2: ```suggestion
including the standard locations for commonly used user directories such as "Desktop"
and "Documents", as well as critical directories more typically used by the operating
```

system itself.

For a full list of the blocked user directories, see [here](https://wicg.github.io/file-system-access/#enumdef-wellknowndirectory).

> Directories _within_ these blocked user directories may still be shared.

Reviewer: ```suggestion
Subdirectories _within_ these blocked user directories may still be shared.
```

> Subdirectories _within_ these blocked user directories may still be shared.

Reviewer: Looks like the docs linter wants `*within*` instead.
